x-pack/plugins/infra/public/pages/metrics/inventory_view/components/waffle/node_square.tsx

```diff
@@ -112,15 +112,14 @@ export const NodeSquare = ({
   color: string;
   nodeName: string;
   value: string;
-  isOverlayOpen: boolean;
+  nodeBorder?: CSSProperties;
```

Comment:
`CSSProperties` makes this prop accept any style attributes, not only `border`. So nothing prevents it from receiving `{ padding-top: "10px }`, for instance. I think we want to only. 

Would this work? 

```tsx
export const NodeSquare = ({
  ...
  showBorder = false
}: {
  ...,
  showBorder?: boolean
}) => {
   const style: CSSProperties | undefined = showBorder ? { border: 'solid 4px #000' } : undefined
   ...
   <SquareOuter color={color} style={style}>
```

then to use it:

```tsx
const nodeSquare = (
    <NodeSquare
      squareSize={squareSize}
      color={color}
      nodeName={node.name}
      value={value}
      showBorder={detailsItemId === node.name || isPopoverOpen}
    />
  );
```

Reply:
Good point, thanks :) I changed it as suggested